### PR TITLE
Rename typing of `global types`

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

While working on #5041 I noticed these did not have the right casing. Global module types should be all caps, right?

### CODE REVIEW

File: pylint/reporters/reports_handler_mix_in.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 if TYPE_CHECKING:
     from pylint.lint.pylinter import PyLinter
 
-ReportsDict = DefaultDict[IChecker, List[Tuple[str, str, Callable]]]
+REPORTS_DICT = DefaultDict[IChecker, List[Tuple[str, str, Callable]]]
```
Comment (cdce8p) [inline pylint/reporters/reports_handler_mix_in.py:15]: Type aliases are usually PascalCase. So the existing ones are correct.

**Review by cdce8p — CHANGES_REQUESTED** [review]
I would recommend not to move forward with this one.

File: pylint/reporters/reports_handler_mix_in.py (hunk 1 above)
Comment (DanielNoord) [reply pylint/reporters/reports_handler_mix_in.py:15]: My bad, I thought this was only for `TypedDict` and `NamedTuple` as they can use `class` to instantiate. Going to close!

**Comment by cdce8p** [comment]
Just for reference: https://docs.python.org/3/library/typing.html#type-aliases

**Pull request closed without merging**